packages/medusa/src/types/product-variant.ts

```diff
@@ -219,3 +221,19 @@ export class ProductVariantPricesCreateReq {
   @IsInt()
   max_quantity?: number
 }
+
+/**
+ * @schema DecoratedVariant
+ * title: "Decorated Product Variant"
+ * type: object
+ * allOf:
+ *   - $ref: "#/components/schemas/PricedVariant"
+ *   - type: object
+ *     properties:
+ *       purchasable:
+ *         type: boolean
+ *         description: Boolean indicating if variant is purchasable.
+ */
+export type DecoratedVariant = PricedVariant & {
```

Comment:
suggestion: Should we call it `PurchasablePricedVariant` and same for the product? Decorated is a bit generic I think and you don't know what it is unless you look at the content of the type

Reply:
We could, I did consider it but decided against it. 

I went with `Decorated` because adding `PurchasablePricedVariant` seems ok, but then adding subsequent decorated props is going to spawn more and more types, especially if you consider cases where we don't decorate everything (everywhere) all of the time. 

We use typescript and that gives insight into the shape of the type which is why I think going with `Decorated` to indicate "this is more than the base datamodel" is okay. 

I'm not opposed to doing `PurchasablePricedVariant` I just don't think it's very scalable wrt. readability and number of type.

Reply:
I understand your point of view, i am fine with keeping it as it is for now and see how it evolves. If as you said it will not be limited to the notion of purshasable with plenty of properties then it is fine. There is pros and cons for both approaches and why we created a pricedVariant type instead of decorated

Reply:
I am in general more in favour of smaller understandable types that you can compose with and that easily retrievable instead of fewer non understandable types that you have to omit/pick if you want to compose. But let’s bring that discussion for another discussion later 😉

Reply:
After discussions with Pat I removed the notion of a `decoratedVariant`/`decoratedProduct` entirely so this wont be an issue